src/preprocessing/passes/non_clausal_simp.h

```diff
@@ -48,6 +49,40 @@ class NonClausalSimp : public PreprocessingPass
 
   /** Learned literals */
   std::vector<Node> d_nonClausalLearnedLiterals;
+
+  /** Preprocess the boolean skeleton using CryptoMinisat
+      Args:
+      - assertionsToPreprocess: Top level assertions
+      - substs_index: the substitution index in the current context
+      Return:
+      whether the boolean skeleton is not satisfiable and the
+      learned unit clauses
+   */
+  std::pair<bool, std::vector<Node>> preprocessByCryptoMinisat(
+      AssertionPipeline* assertionsToPreprocess, unsigned substs_index);
+
+  /** Preprocess the boolean skeleton using CryptoMinisat
+      Args:
+      - assertionsToPreprocess: Top level assertions
+      - substs_index: the substitution index in the current context
+      Return:
+      whether the boolean skeleton is not satisfiable and the
+      learned unit clauses
+   */
+  std::pair<bool, std::vector<Node>> preprocessByCircuitPropagator(
+      AssertionPipeline* assertionsToPreprocess, unsigned substs_index);
+
+  /** Handle the case that a conflict in the boolean skeleton is detected */
+  inline void handleConflictCase(AssertionPipeline* assertionsToPreprocess)
```

Comment:
I would move that into the .cpp file. No need to use inline either, let the compiler decide how to optimize.